src/edu/csus/ecs/pc2/core/util/IJSONTool.java

```diff
@@ -0,0 +1,273 @@
+    /**
+     * Returns the unique group id for a group.  This could be the assigned group id, or
+     * the elementid if no group id is assigned.
+     *
+     * @param group
+     * @return string representation of the group id
+     */
+    public static String getGroupId(Group group) {
+        String id = group.getElementId().toString();
+        if (group.getGroupId() != -1) {
+            id = Integer.toString(group.getGroupId());
+        }
+        return id;
+    }
+
+    /**
+     * Returns organization (institution) number as a string
+     * @param account to use to fetch organization (institution) from
+     * @return string representation of the organization ID (which is a number)
+     */
+    public static String getOrganizationId(Account account) {
+        String id = account.getInstitutionCode();
+        if (id.startsWith("INST-U-")) {
+            id = id.substring(7);
+        }
+        if (id.startsWith("INST-")) {
+            id = id.substring(5);
+        }
+        return id;
+    }
+
+    /**
+     * Returns account ID as a string
+     * @param account to use to the client id from
+     * @return string representation of the ID (which is a number)
+     */
+    public static String getAccountId(Account account) {
+        String id = "" + account.getClientId().getClientNumber();
+        return id;
+    }
+
+    /**
+     * Returns the problem shortname, if present, otherwise the elementid
+     *
+     * @param problem
+     * @return string which is the shortname or elementid
+     */
+    public static String getProblemId(Problem problem) {
+        String id = problem.getElementId().toString();
+        // if we have a problem shortName use it, otherwise default to the internal id
+        String shortName = problem.getShortName();
+        if (shortName != null && !shortName.equals("")) {
+            id = shortName;
+        }
+        return id;
+    }
+
+    /**
+     * Returns the clarification id which is currently the elementid
+     * Someday change to an ordinal?
+     *
+     * @param clarification
+     * @return string which is the id to use for this clarification
+     */
+    public static String getClarificationId(Clarification clar) {
+        String id = clar.getElementId().toString();
+        return id;
+    }
+
+    /**
+     * Returns the clarification answer id which is currently the elementid
+     * Someday change to an ordinal?
+     *
+     * @param clarAns
+     * @return string which is the id to use for this clarification answer
+     */
+    public static String getClarificationAnswerId(ClarificationAnswer clarAns) {
+        String id = clarAns.getElementId().toString();
+        return id;
+    }
+
+    /**
+     * Return wall time for input elapsed time in ms.
+     *
+     * Calculates based on elapsed time plus contest start time
+     *
+     * @param contest
+     * @param elapsedMS
+     *            - elapsed ms when submission submitted
+     * @return wall time for run.
```

Comment:
Again, I don't see anything in this method that necessarily means it is associated with a _run_.

Reply:
Cleared up the comment in next push.